src/directives/touch.ts

```diff
@@ -63,35 +105,39 @@ function inserted (el, { value }, { context }) {
     end: value.end
   }
 
-  const target = value.parent ? el.parentNode : el
+  return {
+    touchstart: (e: TouchEvent) => touchstart(e, wrapper),
+    touchend: (e: TouchEvent) => touchend(e, wrapper),
+    touchmove: (e: TouchEvent) => touchmove(e, wrapper)
+  }
+}
+
+function inserted (el: TouchHTMLElement, binding: TouchVNodeDirective, vnode: VNode) {
+  const value = binding.value
+  const target = (value.parent ? el.parentNode : el) as TouchHTMLElement
   const options = value.options || { passive: true }
 
   // Needed to pass unit tests
   if (!target) return
 
-  const handlers = {
-    touchstart: e => touchstart(e, wrapper),
-    touchend: e => touchend(e, wrapper),
-    touchmove: e => touchmove(e, wrapper)
-  }
-  target._touchHandlers = Object.assign(Object(target._touchHandlers), {
-    [context._uid]: handlers
-  })
+  const handlers = createHandlers(binding.value)
+  target._touchHandlers = Object(target._touchHandlers)
+  target._touchHandlers[(vnode.context as any)._uid] = handlers
+
   for (const eventName of Object.keys(handlers)) {
-    target.addEventListener(eventName, handlers[eventName], options)
+    target.addEventListener(eventName, handlers[eventName as 'touchstart' | 'touchend' | 'touchmove'] as EventListener, options)
   }
 }
 
-function unbind (el, { value }, { context }) {
-  const target = value.parent ? el.parentNode : el
-
+function unbind (el: TouchHTMLElement, binding: TouchVNodeDirective, vnode: VNode) {
+  const target = (binding.value.parent ? el.parentNode : el) as TouchHTMLElement
   if (!target) return
 
-  const handlers = target._touchHandlers[context._uid]
+  const handlers = target._touchHandlers[(vnode.context as any)._uid]
   for (const eventName of Object.keys(handlers)) {
-    target.removeEventListener(eventName, handlers[eventName])
+    target.removeEventListener(eventName, handlers[eventName as 'touchstart' | 'touchend' | 'touchmove'] as EventListener)
```

Comment:
Use the `keys` helper in VIcon (move it to utils) so you don't need so many type assertions.